### PR TITLE
Wip/hughsie/no jcat context setup

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: libjcat/jcat-engine.c [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
-#include "jcat-context.h"
+#include "jcat-context-private.h"
 #include "jcat-engine-private.h"
 
 typedef struct {
 	JcatContext		*context;		/* weak */
 	JcatBlobKind		 kind;
 	JcatEngineVerifyKind	 verify_kind;
-	gchar			*localstatedir;
+	gboolean		 has_setup;
```
Comment (superm1) [inline libjcat/jcat-engine.c]: point of preference; I think this should actually be `done_setup`

File: libjcat/jcat-context.c [hunk 2]
```diff
@@ -50,7 +50,7 @@ static void
 jcat_context_init (JcatContext *self)
 {
 	JcatContextPrivate *priv = GET_PRIVATE (self);
-	priv->localstatedir = g_strdup (JCAT_LOCALSTATEDIR);
+	priv->keyring_path = g_build_filename (JCAT_LOCALSTATEDIR, "lib", PACKAGE_NAME, NULL);
```
Comment (superm1) [inline libjcat/jcat-context.c]: Hmm but `PACKAGE_NAME` is the library name?  So potentially if something fails to set the new keyring path we end up using `/var/lib/libjcat`?  Maybe it's better to require that to be set in the API.

Comment (hughsie) [reply libjcat/jcat-context.c]: It's just a sane default. In jcat-tool I actually used the user datadir as the sane default, maybe that makes sense to use here as well?

File: libjcat/jcat-engine.c (hunk 1 above)
Comment (hughsie) [reply libjcat/jcat-engine.c]: Completely agree, good catch.

File: libjcat/jcat-context.c (hunk 2 above)
Comment (hughsie) [reply libjcat/jcat-context.c]: e.g.

    -       priv->keyring_path = g_build_filename (JCAT_LOCALSTATEDIR, "lib", PACKAGE_NAME, NULL);
    +       priv->keyring_path = g_build_filename (g_get_user_data_dir (), PACKAGE_NAME, NULL);

Comment (superm1) [reply libjcat/jcat-context.c]: Yeah that's a better sane default I think.

Comment (hughsie) [reply libjcat/jcat-context.c]: Okay, fixed.

**Comment by superm1** [comment]
Lgtm

**Pull request merged**